elements/dash-tree/dash-tree.js

```diff
@@ -0,0 +1,436 @@
+/**
+ * A `DashTree` presents a hierarchical list of children. Each node in the tree
+ * is represented as a `DashTreeItem` element. If a `DashTreeItem` contains only
+ * text it is considered an "end node". If a `DashTreeItem` contains children,
+ * it is considered a "parent node". If a `DashTreeItem` is a parent node, its
```

Comment:
I took the terminology from the ARIA authoring practices guide but I can change it to leaf/parent